posthog/email.py

```diff
@@ -221,6 +219,63 @@ def _send_via_smtp(
                 )
 
 
+def sanitize_email_properties(properties: dict[str, Any]) -> dict[str, Any]:
+    """
+    Sanitizes properties that will be used in email templates to prevent HTML injection.
+    This function recursively processes dictionaries, lists, and scalar values to ensure
+    all string values are properly escaped.
+
+    Args:
+        properties: Dictionary of properties to sanitize
+
+    Returns:
+        Sanitized copy of the properties dictionary
+
+    Raises:
+        TypeError: If an unsupported type is encountered
+    """
+    if properties is None:
+        return {}
+
+    # Special keys that should not be sanitized (e.g., URL query parameters)
+    skip_sanitization_keys = ["utm_tags"]
+
+    # Supported types (besides containers like dict and list)
+    supported_types = (str, int, float, bool, type(None), Decimal)
+
+    def sanitize_value(value: Any) -> Any:
+        if isinstance(value, str):
+            return html.escape(value)
+        elif isinstance(value, dict):
+            return {k: sanitize_value(v) for k, v in value.items()}
+        elif isinstance(value, list):
+            return [sanitize_value(item) for item in value]
+        elif hasattr(value, "_meta") and hasattr(value, "pk"):
+            # Handle Django models by converting to string and escaping
+            return html.escape(str(value))
+        elif isinstance(value, Decimal):
+            # Convert Decimal to float for JSON serialization
+            return float(value)
+        elif isinstance(value, int | float | bool | type(None)):
+            # These types are safe as-is
+            return value
+        else:
+            # Raise an error for unsupported types - this is a security measure to prevent uncaught injections
```

Comment:
I went back forth here but opt'd for the safest approach and it another type needs to be supported it can be added.